p2p/netaddress.go

```diff
@@ -176,7 +176,7 @@ func (na *NetAddress) String() string {
 	if na.str == "" {
 		addrStr := na.DialString()
 		if na.ID != "" {
-			addrStr = IDAddressString(na.ID, addrStr)
+			addrStr = removeProtocolIfDefined(IDAddressString(na.ID, addrStr))
```

Comment:
I think what was proposed in the issue is that we fix `removeProtocolIfDefined ` to remove the protocol from any place in the string, not just from the beginning.